test/core/ext/filters/rbac/rbac_service_config_parser_test.cc

```diff
@@ -638,6 +640,138 @@ TEST(RbacServiceConfigParsingTest, StringMatcherBadTypes) {
+      "        ]\n"
+      "      }\n"
+      "    } ]\n"
+      "  } ]\n"
+      "}";
+  ChannelArgs args = ChannelArgs().Set(GRPC_ARG_PARSE_RBAC_METHOD_CONFIG, 1);
+  auto service_config = ServiceConfigImpl::Create(args, test_json);
+  ASSERT_TRUE(service_config.status().ok());
+  const auto* vector_ptr =
+      (*service_config)->GetMethodParsedConfigVector(grpc_empty_slice());
+  ASSERT_NE(vector_ptr, nullptr);
+  auto* parsed_rbac_config = static_cast<RbacMethodParsedConfig*>(
+      ((*vector_ptr)[RbacServiceConfigParser::ParserIndex()]).get());
+  ASSERT_NE(parsed_rbac_config, nullptr);
+  ASSERT_NE(parsed_rbac_config->authorization_engine(0), nullptr);
+  EXPECT_EQ(parsed_rbac_config->authorization_engine(0)->audit_condition(),
+            Rbac::AuditCondition::kOnDeny);
+  EXPECT_EQ(parsed_rbac_config->authorization_engine(0)->num_audit_loggers(),
+            2);
+}
+
+TEST(RbacServiceConfigParsingTest, BadAuditConditionAndLoggersTypes) {
+  const char* test_json =
+      "{\n"
+      "  \"methodConfig\": [ {\n"
+      "    \"name\": [\n"
+      "      {}\n"
+      "    ],\n"
+      "    \"rbacPolicy\": [ {\n"
+      "      \"filter_name\": \"rbac\",\n"
+      "      \"rules\":{\n"
+      "        \"action\":1,\n"
+      "        \"audit_condition\":{},\n"
+      "        \"audit_loggers\":{}\n"
+      "      }\n"
+      "    } ]\n"
+      "  } ]\n"
+      "}";
+  ChannelArgs args = ChannelArgs().Set(GRPC_ARG_PARSE_RBAC_METHOD_CONFIG, 1);
+  auto service_config = ServiceConfigImpl::Create(args, test_json);
+  EXPECT_EQ(service_config.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(service_config.status().message(),
+            "errors validating service config: ["
+            "field:methodConfig[0].rbacPolicy[0].rules.audit_condition "
+            "error:is not a number; "
+            "field:methodConfig[0].rbacPolicy[0].rules.audit_loggers "
+            "error:is not an array]")
+      << service_config.status();
+}
+
+TEST(RbacServiceConfigParsingTest, BadAuditConditionEnum) {
+  const char* test_json =
+      "{\n"
+      "  \"methodConfig\": [ {\n"
+      "    \"name\": [\n"
+      "      {}\n"
+      "    ],\n"
+      "    \"rbacPolicy\": [ {\n"
+      "      \"filter_name\": \"rbac\",\n"
+      "      \"rules\":{\n"
+      "        \"action\":1,\n"
+      "        \"audit_condition\":100\n"
+      "      }\n"
+      "    } ]\n"
+      "  } ]\n"
+      "}";
+  ChannelArgs args = ChannelArgs().Set(GRPC_ARG_PARSE_RBAC_METHOD_CONFIG, 1);
+  auto service_config = ServiceConfigImpl::Create(args, test_json);
+  EXPECT_EQ(service_config.status().code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(service_config.status().message(),
+            "errors validating service config: ["
+            "field:methodConfig[0].rbacPolicy[0].rules.audit_condition "
+            "error:unknown audit condition]")
+      << service_config.status();
+}
+
+TEST(RbacServiceConfigParsingTest, BadAuditLoggerObject) {
```

Comment:
Let's also test the cases of an unsupported audit logger type and a supported audit logger type with a bad config.  (That latter case will probably require inserting a custom type into the registry.)

Reply:
Added a test logger factory to verify config as well as rejecting bad configs. PTAL.